shared_model/validators/validators_common.cpp

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "validators/validators_common.hpp"
+
+#include <regex>
+
+namespace shared_model {
+  namespace validation {
+
+    bool validateHexString(const std::string &str) {
+      static const std::regex hex_regex{R"([0-9a-fA-F]*)"};
```

Comment:
So we do not check the length of the hex string in this function, although it is used directly to validate hashes and public keys. I would suggest to add specific functions that check our hash and public key formats to this file and use this function internally, like
```c++
bool validatePublicKeyString(const std::string &pk) {
    return pk.size() == crypto::kPublicKeyLength && validateHexString(pk);
}

bool validateHashString(const std::string &hash) {
    return hash.size() == crypto::kHashLength && validateHexString(hash);
}
```
and use them to validate both content and size.

Reply:
PS: it seems now the length constants are defined in `shared_model/cryptography/ed25519_sha3_impl/crypto_provider.hpp` and initialized here:
https://github.com/hyperledger/iroha/blob/34bf8704d0dc8649a0a24ee0f75351d50e64094c/shared_model/cryptography/ed25519_sha3_impl/crypto_provider.cpp#L45-L49
So seems they can be accessed through `shared_model::crypto::DefaultCryptoAlgorithmType`.
Just do not forget that a byte in blob is two chars in hex string!

Reply:
This is checked in business logic. Particularly field_validator does that. In protobuf checker we only check if protobuf structure is okay. Size of the public key and hashes is checked later

Reply:
I do not see a difference between checking the string content and string size: why we check one and dismiss the other?